cloud/scope/powervs_machine.go

```diff
@@ -135,6 +138,16 @@ func NewPowerVSMachineScope(params PowerVSMachineScopeParams) (scope *PowerVSMac
 		return
 	}
 
+	for _, rcEndpoint := range params.ServiceEndpoint {
+		if rcEndpoint.ServiceID == endpoints.RCServiceID {
+			errr := rc.SetServiceURL(rcEndpoint.URL)
```

Comment:
and also wondering if we can move this function to endpoint package as utility function so that can be reused everywhere?

Reply:
Noted, will do the required changes.